src/solver/simulation/sim_alloc_probleme_hebdo.cpp

```diff
@@ -580,259 +535,82 @@ void SIM_DesallocationProblemeHebdo(PROBLEME_HEBDO& problem)
-            delete[] problem.PaliersThermiquesDuPays[k]
-                      ->PuissanceDisponibleEtCout[j]
-                      ->PuissanceMinDuPalierThermique;
-            delete[] problem.PaliersThermiquesDuPays[k]
-                      ->PuissanceDisponibleEtCout[j]
-                      ->PuissanceMinDuPalierThermique_SV;
-            delete[] problem.PaliersThermiquesDuPays[k]
-                      ->PuissanceDisponibleEtCout[j]
-                      ->NombreMaxDeGroupesEnMarcheDuPalierThermique;
-            delete[] problem.PaliersThermiquesDuPays[k]
-                      ->PuissanceDisponibleEtCout[j]
-                      ->NombreMinDeGroupesEnMarcheDuPalierThermique;
-            delete[] problem.PaliersThermiquesDuPays[k]
-                      ->PuissanceDisponibleEtCout[j]
-                      ->CoutHoraireDuPalierThermiqueUp;
-            delete[] problem.PaliersThermiquesDuPays[k]
-                      ->PuissanceDisponibleEtCout[j]
-                      ->CoutHoraireDuPalierThermiqueDown;
-            delete problem.PaliersThermiquesDuPays[k]->PuissanceDisponibleEtCout[j];
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->CoutHoraireDeProductionDuPalierThermique);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->CoutHoraireDeProductionDuPalierThermiqueRef);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->PuissanceDisponibleDuPalierThermique);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->PuissanceDisponibleDuPalierThermiqueRef);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->PuissanceDisponibleDuPalierThermiqueRef_SV);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->PuissanceMinDuPalierThermique);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->PuissanceMinDuPalierThermique_SV);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->NombreMaxDeGroupesEnMarcheDuPalierThermique);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->NombreMinDeGroupesEnMarcheDuPalierThermique);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
+                      ->CoutHoraireDuPalierThermiqueUp);
+            MemFree(problem.PaliersThermiquesDuPays[k]
+                      .PuissanceDisponibleEtCout[j]
```

Comment:
Be careful, you're mixing `new` (C++) and `free` (C)

Reply:
Will be deleted when replaced with vectors